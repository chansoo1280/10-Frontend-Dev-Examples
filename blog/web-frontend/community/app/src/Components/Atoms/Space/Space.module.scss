.space {
    display: flex;
    align-items: center;
    padding: 8px 12px;
    gap: 8px;

    &--small {
        padding: 0;
        gap: 0px;
    }

    &--large {
        padding: 16px 24px;
        gap: 24px;
    }

    &--vertical {
        flex-direction: column;
    }

    &__box {
        flex: 1;
    }
}