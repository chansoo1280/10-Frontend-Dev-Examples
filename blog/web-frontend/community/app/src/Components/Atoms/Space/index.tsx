// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { defaultProps } from "@Components"
import styles from "./Space.module.scss"
// #endregion Local Imports

interface SpaceProps extends defaultProps {
    direction?: "vertical"
    size?: "small" | "medium" | "large"
    gap?: number
    padding?: string | number
}
const Space = (props: SpaceProps): JSX.Element => {
    const { direction, size, show, className, gap, padding, ...rest } = props
    const prefixCls = "space"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
            [styles[`${prefixCls}--${size}`]]: size,
            [styles[`${prefixCls}--${direction}`]]: direction,
        },
        className,
    )
    return <div {...rest} className={classes} style={{ gap, padding }} />
}
const Box = (props: defaultProps) => {
    const { ...rest } = props
    return <div className={classNames(styles[`space__box`])} {...rest} />
}
Space.Box = Box
export default Space
