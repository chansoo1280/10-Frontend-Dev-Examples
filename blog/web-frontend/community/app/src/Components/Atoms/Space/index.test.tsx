import { render, screen } from "@testing-library/react"
import Space from "./"
describe("Space", () => {
    it("renders space unchanged", () => {
        const { container } = render(
            <Space>
                <div>1</div>
                <div>2</div>
                <Space.Box>3</Space.Box>
            </Space>,
        )
        expect(container).toMatchSnapshot()
    })
    it("should render a space with the class", () => {
        render(
            <Space data-testid="space">
                <div>1</div>
                <div>2</div>
                <Space.Box>3</Space.Box>
            </Space>,
        )

        const space = screen.getByTestId("space")
        expect(space).toBeInTheDocument()
        expect(space).toHaveClass("space")
    })
    it("should render a space with the class of virtical", () => {
        render(
            <Space direction="vertical" data-testid="space">
                <div>1</div>
                <div>2</div>
                <Space.Box>3</Space.Box>
            </Space>,
        )

        const space = screen.getByTestId("space")
        expect(space).toBeInTheDocument()
        expect(space).toHaveClass("space", "space--vertical")
    })
})
