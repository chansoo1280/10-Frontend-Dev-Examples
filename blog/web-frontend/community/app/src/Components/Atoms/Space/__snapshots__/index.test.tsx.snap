// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space renders space unchanged 1`] = `
<div>
  <div
    class="space"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div
      class="space__box"
    >
      3
    </div>
  </div>
</div>
`;
